compute_sdk/tests/unit/test_client.py

```diff
@@ -331,6 +331,36 @@ def test_register_function_deprecated_args(gcc, dep_arg):
     assert dep_arg in str(warning)
 
 
+def test_register_function_docstring(gcc):
+    gcc.web_client = mock.MagicMock()
+
+    def func_with_docstring():
+        """This is a docstring"""
+        return "Funky"
+
+    gcc.register_function(func_with_docstring)
+
+    a, _ = gcc.web_client.register_function.call_args
+    func_data = a[0]
+    assert func_data.description == "This is a docstring"
+
+
+def test_register_function_multiline_dosctring(gcc):
+    gcc.web_client = mock.MagicMock()
+
+    def func_with_multiline_docstring():
+        """This is a docstring
+        that spans multiple lines
+        """
+        return "Funky"
+
+    gcc.register_function(func_with_multiline_docstring)
+
+    a, _ = gcc.web_client.register_function.call_args
+    func_data = a[0]
+    assert func_data.description == "This is a docstring\nthat spans multiple lines"
```

Comment:
These two look to me to be fundamentally the same test.  Consider:
```python
@pytest.mark.parametrize("docstring", (
    "This is a docstring",
    "This is a docstring\nthat spans multiple lines",
))
def test_register_function_docstring(gcc, docstring):
    gcc.web_client = mock.Mock(spec=WebClient)

    def test_func_docstring():
        return "Funky"
    test_func_docstring.__doc__ = docstring
    ...
    assert func_data.description == docstring
```

Reply:
there's a subtle distinction here - the docstring in the second test is actually `"This is a docstring\n        that spans multiple lines"`. the intent of the second test is to verify the dedenting, with a relatively natural and innocuous docstring.

clearly that intent isn't obvious - should I add a comment, rename the test, both, something else?

Reply:
Ah; I suppose I consider that a basal detail "that everyone knows."

Okay, if that's your concern, then I might consider stating precisely that in the tested string.  Another option might be to add another item in the parameter list:

```python
@pytest.mark.parametrize("docstring", (
    "This is a docstring",
    "This is a docstring\nthat spans multiple lines",
    "This is a docstring\n    that should get dedented\n    by the runtime"
))
```

But I haven't verified that locally, so let me know what works best.

Reply:
per side conversation, changing this to parametrize over a set of functions with various docstrings